app/public/js/playlists/playlistsCtrl.js

```diff
@@ -129,4 +130,32 @@ app.controller('PlaylistsCtrl', function (
             return newSize;
         }
     };
+
+    /**
+     * Responsible to shuffle a playlist
+     * instead of having to play then shuffle
+     * @param tracks [ tracks in playlist ]
+     * @method shufflePlaylist
+     */
+    $scope.shufflePlaylist = function (tracks){
```

Comment:
This seems unnecessary because the shuttle should in theory do what you trying to accomplish without modifying the list order.